umbraco/Umbraco-CMS
# Ability to configure a remote URL for displaying different content on the content dashboard

## Details 

Adds the `ContentDashboardBaseUrl` appSetting in the Web.config file, so the intro dashboard ("Getting Started") can be used to display different content on Umbraco Cloud. The change will affect all new projects. In the case that the config setting is not present the default content-endpoint will be called, therefore not changing any existing behaviour.

Additionally, more information is reported from the existing dashboard - a `hosting` parameter is added by the end to indicate if the solution is running on Umbraco Cloud.
- If that is the case **Umbraco.Deploy.Cloud.dll** will exist


## Test:
1. Test 1
   1. Before switching to this branch, check what content is being displayed on the "Getting Started" dashboard.
   2. Switch to this branch and access the dashboard again, verifying that nothing has changed.
2. Test 2
   1.  Check that since you don't have _Umbraco.Deploy.Cloud.dll_ in **...src\Umbraco.Web.UI\bin** the value of the `hosting` parameter is `onpremises` - mimicking installation that is not on Umbraco Cloud.
3. Test 3
   1. Change the value of _Umbraco.Deploy.Cloud.dll_ (inside `IOHelper.IsOnCloud`) to a dll that is present in **...src\Umbraco.Web.UI\bin** & verify that the `hosting` value is `cloud`.
   2. Add `<add key="Umbraco.Core.ContentDashboardBaseUrl" value="https://dashboard.umbraco.org/"/>` to Web.config and check the content persists. (later on, this endpoint should be changed but for the moment, we don't have a different one for testing)
